server/webapp/WEB-INF/rails.new/webpack/views/agents/agent_row_widget.js.msx

```diff
@@ -61,9 +61,14 @@ const AgentRowWidget = {
     const environments    = joinOrNoneSpecified(agent.environments());
     let agentStatus       = agent.status();
     const isBuildingAgent = !agent.buildDetails().isEmpty();
+    const isElasticAgent  = agent.isElasticAgent();
     let selectAgentCheckbox;
     let hostnameLink      = <span>{agent.hostname()}</span>;
 
+    if (isElasticAgent) {
+      agent.fetchElasticAgentIcon();
```

Comment:
This could be dangerous with hundreds of agents being polled every 10 seconds. For such things, it's better to pull down the plugin infos object via `PluginInfos.all()` in `single_page_apps/agents.js`. Sort of like we do it [here](https://github.com/gocd/gocd/blob/master/server/webapp/WEB-INF/rails.new/webpack/single_page_apps/elastic_profiles.js#L29-L48)

This plugin infos object can be passed onto this same view where you can decide which icon you want to render.